src/lib.rs

```diff
@@ -99,96 +103,111 @@ impl Superoptimizer {
         let module = Module::from_bytes(&self.spec).unwrap();
 
         // TODO(taegyunkim): Use num_cpus crate to appropriately set the number of workers.
-        let num_workers = 1;
-        let mut candidates: Vec<wasm::Candidate> = Vec::with_capacity(num_workers);
+        let num_workers = num_cpus::get();
+        let candidates = Arc::new(Mutex::new(Vec::with_capacity(num_workers)));
```

Comment:
if the main thread handles processing candidates, this doesn't need to be Arc. Use res_sender to send None or candidate? instead of sending ()?